pkg/volume/azure_dd/vhd_util.go

```diff
@@ -56,41 +84,39 @@ func findDiskByLun(lun int, io ioHandler, exe exec.Interface) (string, error) {
 			if len(arr) < 4 {
 				continue
 			}
-			target, err := strconv.Atoi(arr[0])
+			l, err := strconv.Atoi(arr[3])
 			if err != nil {
-				glog.Errorf("failed to parse target from %v (%v), err %v", arr[0], name, err)
+				// unknown path format, continue to read the next one
+				glog.Errorf("failed to parse lun from %v (%v), err %v", arr[3], name, err)
 				continue
 			}
-
-			// as observed, targets 0-3 are used by OS disks. Skip them
-			if target > 3 {
-				l, err := strconv.Atoi(arr[3])
+			if lun == l {
+				// find the matching LUN
+				// read vendor and model to ensure it is a VHD disk
+				vendor := path.Join(sys_path, name, "vendor")
+				model := path.Join(sys_path, name, "model")
+				out, err := exe.Command("cat", vendor, model).CombinedOutput()
 				if err != nil {
-					// unknown path format, continue to read the next one
-					glog.Errorf("failed to parse lun from %v (%v), err %v", arr[3], name, err)
+					glog.Errorf("failed to cat device vendor and model, err: %v", err)
 					continue
 				}
-				if lun == l {
-					// find the matching LUN
-					// read vendor and model to ensure it is a VHD disk
-					vendor := path.Join(sys_path, name, "vendor")
-					model := path.Join(sys_path, name, "model")
-					out, err := exe.Command("cat", vendor, model).CombinedOutput()
-					if err != nil {
-						glog.Errorf("failed to cat device vendor and model, err: %v", err)
-						continue
-					}
-					matched, err := regexp.MatchString("^MSFT[ ]{0,}\nVIRTUAL DISK[ ]{0,}\n$", strings.ToUpper(string(out)))
-					if err != nil || !matched {
-						glog.V(4).Infof("doesn't match VHD, output %v, error %v", string(out), err)
-						continue
+				matched, err := regexp.MatchString("^MSFT[ ]{0,}\nVIRTUAL DISK[ ]{0,}\n$", strings.ToUpper(string(out)))
+				if err != nil || !matched {
+					glog.V(4).Infof("doesn't match VHD, output %v, error %v", string(out), err)
+					continue
+				}
+				// find a disk, validate name
+				dir := path.Join(sys_path, name, "block")
+				if dev, err := io.ReadDir(dir); err == nil {
+					found := false
+					for _, diskName := range azureDisks {
+						glog.V(12).Infof("validating disk %q with sys disk %q", dev[0].Name(), diskName)
+						if string(dev[0].Name()) == diskName {
+							found = true
+							break
+						}
 					}
-					// find it!
-					dir := path.Join(sys_path, name, "block")
-					dev, err := io.ReadDir(dir)
-					if err != nil {
-						glog.Errorf("failed to read %s", dir)
-					} else {
+					if !found {
```

Comment:
got it, you are right. Maybe add a little comments indicating you are validate that the disks are not belong to system disk...